ops/_private/harness.py

```diff
@@ -3863,7 +3863,7 @@ def _notice_matches(
         # For example: if user_id filter is set and it doesn't match, return False.
         if user_id is not None and not (notice.user_id is None or user_id == notice.user_id):
             return False
-        if types is not None and notice.type not in types:
+        if types is not None and notice.type.value not in types:
             return False
```

Comment:
`types` is a collection of `str`s. I think this is mimicking the logic of `pebble`, so I think it makes sense for us to use `notice.type.value` here. Was this always silently broken? Or silently worked because `notice.type` was often a `str` -- while now it's supposed to always be an `enum` member?

Reply:
I think you can answer that question, as notices are not really used by charmers yet.
Which means that the question can be rephrased as "do we have a test for this harness feature?" and "does that test actually work?".